internal/controller/controllers/cluster_controller.go

```diff
@@ -171,17 +228,121 @@ func (r *ClusterReconciler) updateStatus(ctx context.Context, cluster *adiiov1al
+			r.Log.Infof("updating %s != %s", old, new)
+		}
+	}
+
+	updateString(spec.Name, c.Name, &params.Name)
+
+	if spec.OpenshiftVersion != c.OpenshiftVersion {
+		return false, ctrl.Result{}, errors.Errorf("Openshift version cannot be updated")
+	}
+
+	updateString(spec.BaseDNSDomain, c.BaseDNSDomain, &params.BaseDNSDomain)
+	updateString(spec.ClusterNetworkCidr, c.ClusterNetworkCidr, &params.ClusterNetworkCidr)
+
+	if spec.ClusterNetworkHostPrefix != c.ClusterNetworkHostPrefix {
+		params.ClusterNetworkHostPrefix = swag.Int64(spec.ClusterNetworkHostPrefix)
+		update = true
+	}
+
+	updateString(spec.ServiceNetworkCidr, c.ServiceNetworkCidr, &params.ServiceNetworkCidr)
+	updateString(spec.APIVip, c.APIVip, &params.APIVip)
+	updateString(spec.APIVipDNSName, swag.StringValue(c.APIVipDNSName), &params.APIVipDNSName)
+	updateString(spec.IngressVip, c.IngressVip, &params.IngressVip)
+	updateString(spec.MachineNetworkCidr, c.MachineNetworkCidr, &params.MachineNetworkCidr)
+	updateString(spec.SSHPublicKey, c.SSHPublicKey, &params.SSHPublicKey)
+
+	if spec.VIPDhcpAllocation != swag.BoolValue(c.VipDhcpAllocation) {
+		params.VipDhcpAllocation = swag.Bool(spec.VIPDhcpAllocation)
+	}
+
+	updateString(spec.HTTPProxy, c.HTTPProxy, &params.HTTPProxy)
+	updateString(spec.HTTPSProxy, c.HTTPSProxy, &params.HTTPSProxy)
+	updateString(spec.NoProxy, c.NoProxy, &params.NoProxy)
+
+	if spec.UserManagedNetworking != swag.BoolValue(c.UserManagedNetworking) {
+		params.UserManagedNetworking = swag.Bool(spec.UserManagedNetworking)
+	}
+
+	updateString(spec.AdditionalNtpSource, c.AdditionalNtpSource, &params.AdditionalNtpSource)
+
+	// TODO: handle InstallConfigOverrides
+
+	data, err := r.getPullSecret(ctx, spec.PullSecretRef.Name, spec.PullSecretRef.Namespace)
+	if err != nil {
+		return false, ctrl.Result{}, errors.Wrap(err, "failed to get pull secret for update")
+	}
+	updateString(data, c.PullSecret, &params.PullSecret)
+
+	if !update {
+		return update, ctrl.Result{}, nil
+	}
+
+	updatedCluster, err := r.Installer.UpdateClusterInternal(ctx, installer.UpdateClusterParams{
+		ClusterUpdateParams: params,
+		ClusterID:           *c.ID,
+	})
+
+	if err != nil {
+		return update, ctrl.Result{}, errors.Wrap(err, "failed to update cluster")
+	}
+
+	r.Log.Infof("Updated cluster %s %s", cluster.Name, cluster.Namespace)
+	reply, err := r.updateState(ctx, cluster, updatedCluster)
+	return update, reply, err
+
 }
 
 func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	mapSecretToCluster := handler.ToRequestsFunc(
+		func(a handler.MapObject) []reconcile.Request {
+			clusters := &adiiov1alpha1.ClusterList{}
+			if err := r.List(context.Background(), clusters); err != nil {
+				return []reconcile.Request{}
+			}
+			reply := make([]reconcile.Request, 0, len(clusters.Items))
+			for _, cluster := range clusters.Items {
+				if cluster.Spec.PullSecretRef.Name == a.Meta.GetName() &&
+					cluster.Spec.PullSecretRef.Namespace == a.Meta.GetNamespace() {
```

Comment:
general question, can `PullSecretRef.Namespace` be diffrent than `cluster.Namespace`?

Reply:
I asked @mhrivnak the same question, looks like it may happen. i'm not sure if we need to add a different permissions in this  case.